packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -98,19 +102,25 @@ const getBase64Image = imageProps => {
     })
   }
 
-  const promise = new Promise((resolve, reject) => {
-    base64Img.requestBase64(requestUrl, (a, b, body) => {
-      // TODO: against dogma, confirm whether writeFileSync is indeed slower
-      fs.promises
-        .writeFile(cacheFile, body)
-        .then(() => resolve(body))
-        .catch(e => {
-          console.error(
-            `Contentful:getBase64Image: failed to write ${body.length} bytes remotely fetched from \`${requestUrl}\` to: \`${cacheFile}\`\nError: ${e}`
-          )
-          reject(e)
-        })
+  const promise = new Promise(async (resolve, reject) => {
```

Comment:
There shouldn't be a need to wrap this all in a Promise. It's an async function so it returns a promise anyway, and you can just await the fs-extra calls.

Reply:
We need the promise for the caching layer. Otherwise I'd have removed it

Reply:
See: https://github.com/gatsbyjs/gatsby/pull/29034/files#diff-55db4898b8c61cf48dade29472a46c02900ba554718a7fe91cc2460a61235e26R126

Reply:
Yeah, but async fucntions return promises anyway, so no need to wrap it like you've done.

Reply:
```js
async function loadImage() {
```

..then:

```js
const promise = loadImage()
```

Reply:
Well yeah I can make a small function that is called. I tried not to alter the code structure to much.

Reply:
@ascorbic done